### ISSUE TITLE
SharedClusterSnapshotRestoreIT#testParallelRestoreOperationsFromSingleSnapshot fails on Windows

### ISSUE THREAD

**Issue by andrershov** [elastic/elasticsearch#44630]
There are frequent test failures on Windows runs, with the following error message:
```
RemoteTransportException[[node_sm0][127.0.0.1:63269][cluster:admin/snapshot/restore]]; nested: RepositoryException[[test-restore-snapshot-repo] could not read repository data from index blob]; nested: AccessDeniedException[C:\Users\jenkins\workspace\elastic+elasticsearch+6.8+multijob-windows-compatibility\os\windows-2012-r2\server\build\testrun\integTest\J1\temp\org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT_68358DAEA649AB85-001\tempDir-002\repos\fWGBJNmGFZ\pending-incompatible-snapshots-ZX9868kzR4iwcBU9gL50BQ -> C:\Users\jenkins\workspace\elastic+elasticsearch+6.8+multijob-windows-compatibility\os\windows-2012-r2\server\build\testrun\integTest\J1\temp\org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT_68358DAEA649AB85-001\tempDir-002\repos\fWGBJNmGFZ\incompatible-snapshots];
```
For example, https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob-windows-compatibility/os=windows-2012-r2/16/console

**Comment by original-brownbear** [comment]
Seems like this might be a more systemic issue since we also have https://github.com/elastic/elasticsearch/issues/44558 ... if I remember correctly you can get this exception with an atomic move like we do here if some other operation is currently reading the directory the target is in which I guess could be the case in this specific test ... I'll have to brush up on my Windows to see how this can be fixed :)

**Comment by ebadyano** [comment]
another failure:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+multijob-windows-compatibility/os=windows-2012-r2/23/consoleFull
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+multijob-windows-compatibility/os=windows-2012-r2/14/consoleFull

**Comment by original-brownbear** [comment]
This is fixed in 7.4+ and muted in all earlier versions now => closing
